### PR TITLE
Fix System.Net.Primitives tests

### LINKED ISSUE #7906: Building System.Net.Primitives.FunctionalTests against packages fails
Building the FunctionalTests with /p:BuildTestsAgainstPackages=true throws errors. Currently <KeepProjectReference>true</KeepProjectReference> has been added to keep the project reference.
We should be able to remove this project reference (just reference the nupkg) and build the test project without errors.

### PR DESCRIPTION
**Pull request by davidsh**
Clean up System.Net.Primitives unit and functional tests. Moved some
functional tests to the unit tests projects since they are testing
non-public methods.

Change the `public` attribute on some of the CookieContainer` methods to
match the public ref contract. This wasn't done properly when this code
was first ported from .NET Framework to .NET Core. Some of the public
methods are only public in Desktop and not Core.

With these changes, the System.Net.Primitives tests can now build and
run with `/p:BuildTestsAgainstPackages=true`.

Fixes #7906.

### CODE REVIEW

**Comment by davidsh** [comment]
cc: @stephentoub @ericstj @weshaggard @CIPop

**Comment by davidsh** [comment]
CI Failure:
http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/master/job/windows_nt_debug_prtest/721/
Test Result (1 failure / +1)
System.Net.Http.Functional.Tests.PostScenarioTest.PostUsingChunkedEncoding_Success(serverUri: https://corefx-net.cloudapp.net/Echo.ashx)

**Comment by davidsh** [comment]
Test Innerloop Windows_NT Debug Build and Test

File: src/System.Net.Primitives/tests/UnitTests/CookieContainerTest.cs [hunk 1]
```diff
@@ -3,19 +3,229 @@
...
+        private static readonly Cookie c2 = new Cookie("name2", "value", "", u1) { Secure = true };
+        private static readonly Cookie c3 = new Cookie("name3", "value", "", u1) { Port = "\"80, 90, 100, 443\"" };
+        private static readonly Cookie c4 = new Cookie("name4", "value", "", u2);
+        private static readonly Cookie c5 = new Cookie("name5", "value", "/path", u2);
+        private static readonly Cookie c6 = new Cookie("name6", "value", "", "." + u3);
+        private static readonly Cookie c7 = new Cookie("name7", "value", "", "." + u3);
+        private static readonly Cookie c8 = new Cookie("name8", "value");
+        private static readonly Cookie c9 = new Cookie("name9", "value");
+        private static readonly Cookie c10 = new Cookie("name10", "value");
+        private static readonly Cookie c11 = new Cookie("name11", "value") { Port = "\"80, 90, 100\"" };
```
Comment (stephentoub) [inline src/System.Net.Primitives/tests/UnitTests/CookieContainerTest.cs]: Nit: normally we'd name these static fields with an ```s_``` prefix

**Comment by stephentoub** [comment]
LGTM

File: src/System.Net.Primitives/tests/UnitTests/CookieContainerTest.cs (hunk 1 above)
Comment (davidsh) [inline src/System.Net.Primitives/tests/UnitTests/CookieContainerTest.cs]: I was only planning on moving source lines around and not do too much code formatting cleanup.  But I will fix this.

Comment (davidsh) [inline src/System.Net.Primitives/tests/UnitTests/CookieContainerTest.cs]: On second thought, there are other formatting changes we need to do so, I'm going to defer that to another PR.

**Pull request merged**